### ISSUE TITLE
Add "Insert type annotation" intention

### ISSUE THREAD

**Issue by Daenyth** [carymrobbins/intellij-haskforce#215]
For example,

```haskell
foo x = 1 + x
```

With cursor on `foo`, hit <kbd>alt+enter</kbd>, "Insert type annotation", would result in

```haskell
foo :: Num a => a -> a
foo x = 1 + x
```

**Comment by carymrobbins** [comment]
Are you using ghc-mod?  If so, you should get that intention, unless you are using `-fno-warn-missing-signatures`.

If you are not using ghc-mod, see the [Quick Start Guide](https://github.com/carymrobbins/intellij-haskforce/wiki/Quick-Start-Guide#configuring-external-tools) for help with configuring it for HaskForce.

**Comment by Daenyth** [comment]
I believe I am. I have it installed via stack and the path is configured, I see a version number detected in the config panel.

<img width="771" alt="screen shot 2015-12-06 at 7 10 05 pm" src="https://cloud.githubusercontent.com/assets/14644/11616593/028385d8-9c4d-11e5-91ee-a143674ea537.png">

I don't see any intentions for haskell at all; in contrast I have a top-level tree of scala intentions from the scala plugin

<img width="629" alt="screen shot 2015-12-06 at 7 10 48 pm" src="https://cloud.githubusercontent.com/assets/14644/11616598/14d53a6a-9c4d-11e5-8484-0bdc4f1c6d6d.png">

**Comment by Daenyth** [comment]
I just restarted intellij and saw this:

```
7:17:39 PM ghc-modi error
           Unable to parse problems from ghc-modi: Warning: ghc-modi is deprecated please use 'ghc-mod legacy-interactive' instead
           ghc-modi: ghc-mod executable on PATH has different version, check your installation!
           ghc-mod: cabal: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
           Configure
7:17:39 PM ghc-modi error
           Killing ghc-modi due to process failure.
           You can restart it using Tools > Restart ghc-modi
           Executing ghc-modi command 'browse -d Prelude' failed with error: Failed to write command to ghc-modi: 
           Configure (show balloon)
```

**Comment by carymrobbins** [comment]
@Daenyth - Configure ghc-modi to use the `ghc-mod` executable with the `legacy-interactive` flag.

**Comment by Daenyth** [comment]
Is this correct? It still doesn't seem to be working.

<img width="694" alt="screen shot 2015-12-06 at 9 16 30 pm" src="https://cloud.githubusercontent.com/assets/14644/11617554/c8de77fe-9c5e-11e5-9d42-f180130740d6.png">

I get this now;

```
9:18:34 PM ghc-modi error
           Unable to parse problems from ghc-modi: ghc-mod: cabal: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
           Configure
9:18:34 PM ghc-modi error
           Killing ghc-modi due to process failure.
           You can restart it using Tools > Restart ghc-modi
           Executing ghc-modi command 'browse -d Prelude' failed with error: Failed to write command to ghc-modi: 
           Configure (show balloon)
```

My Tools menu doesn't have a `Restart ghc-modi` entry. I'm on OSX if that makes a difference.

 Sorry if I'm missing something obvious; I'm just getting started with Haskell and I'm not familiar with the tooling

**Comment by Daenyth** [comment]
Maybe related to this: https://github.com/kazu-yamamoto/ghc-mod/issues/660

**Comment by carymrobbins** [comment]
Yep, I bet you're right.  Do you have a `ghc` on your path?

**Comment by Daenyth** [comment]
I do in my shell. I'm not sure how to get the intellij .app to include it.

My GHC sdk is set up with that directory;

<img width="689" alt="screen shot 2015-12-06 at 9 42 00 pm" src="https://cloud.githubusercontent.com/assets/14644/11617888/488f7040-9c62-11e5-9e0e-bf1cea461857.png">

**Comment by Daenyth** [comment]
I tried adjusting my path with instructions from here: http://apple.stackexchange.com/questions/51677/how-to-set-path-for-finder-launched-applications

Now I get this
```
ghc-modi error
           Unable to parse problems from ghc-modi: [1 of 5] Compiling CabalHelper.Types ( CabalHelper/Types.hs, /Users/daenyth/.ghc-mod/cabal-helper/CabalHelper/Types.o )
           [2 of 5] Compiling CabalHelper.Common ( CabalHelper/Common.hs, /Users/daenyth/.ghc-mod/cabal-helper/CabalHelper/Common.o )
           [3 of 5] Compiling CabalHelper.Sandbox ( CabalHelper/Sandbox.hs, /Users/daenyth/.ghc-mod/cabal-helper/CabalHelper/Sandbox.o )
           [4 of 5] Compiling CabalHelper.Licenses ( CabalHelper/Licenses.hs, /Users/daenyth/.ghc-mod/cabal-helper/CabalHelper/Licenses.o )
           [5 of 5] Compiling Main             ( CabalHelper/Main.hs, /Users/daenyth/.ghc-mod/cabal-helper/Main.o )
           Linking /Users/daenyth/.ghc-mod/cabal-helper/cabal-helper-0.6.2.0-Cabal-1.22.4.0 ...
```

**Comment by carymrobbins** [comment]
Ok, I just need to have the `GhcMod.parseProblem` method ignore compiling output.

It seems like, since it wasn't an error from ghc-mod, the interactive process should still run freely.  Therefore, once it's done compiling, stuff should start working properly.

I'll work on updating that method so you don't get that error.

**Comment by Daenyth** [comment]
Thanks!

**Comment by Daenyth** [comment]
That said, I don't have intentions still. Intellij 14 ultimate version, if that makes a difference.

**Comment by carymrobbins** [comment]
@Daenyth - Do you have error highlighting and/or the ability to show type info?  I also wonder if it has something to do with building against IntelliJ 15.

**Comment by carymrobbins** [comment]
Confirmed that it works in IntelliJ 14.

Also, note that the intentions aren't actually listed in the **Editor > Intentions** settings.  If that's the problem, I've opened https://github.com/carymrobbins/intellij-haskforce/issues/217 to address that.

**Comment by Daenyth** [comment]
Quick doc does show me type signatures, though I saw it crash once - I didn't grab the exception unfortunately. InputMismatchException I think.

I don't seem to have error highlighting; I put in a `foo x = x + 1 ++ "hi"` and it's not marked as an error; quick doc tells me `foo` is `[Char] -> [Char]`.

I'm thinking the intentions are missing because when I hit alt+enter on a function name no menu comes up.

**Comment by carymrobbins** [comment]
Could you check the IntelliJ logs?  You can find them here - https://intellij-support.jetbrains.com/hc/en-us/articles/206827517-Locating-IDE-log-files

I recommend restarting IntelliJ and running `tail -f <path/to/idea.log>` in a terminal so you can watch the output as stuff is happening.  If you see any errors or tracebacks, feel free to link to them (it may be convenient to paste them into a [gist](https://gist.github.com/)).

**Comment by Daenyth** [comment]
I opened the project, and did the Quick Doc action on `foo`, logs show this

```
2016-01-14 06:17:47,023 [ 240286]   WARN - ng.annotation.external.GhcModi - java.util.InputMismatchException
java.util.concurrent.ExecutionException: java.util.InputMismatchException
	at java.util.concurrent.FutureTask$Sync.innerGet(FutureTask.java:232)
	at java.util.concurrent.FutureTask.get(FutureTask.java:91)
	at com.haskforce.highlighting.annotation.external.GhcModi.getFuture(GhcModi.java:68)
	at com.haskforce.highlighting.annotation.external.GhcModi.getFutureType(GhcModi.java:60)
	at com.haskforce.highlighting.annotation.external.TypeInfoUtil.getTypeInfo(TypeInfoUtil.java:66)
	at com.haskforce.highlighting.annotation.HaskellDocumentationProvider.generateDoc(HaskellDocumentationProvider.java:43)
	at com.intellij.lang.documentation.CompositeDocumentationProvider.generateDoc(CompositeDocumentationProvider.java:134)
	at com.intellij.codeInsight.documentation.DocumentationManager$DefaultDocumentationCollector$3.compute(DocumentationManager.java:1113)
	at com.intellij.codeInsight.documentation.DocumentationManager$DefaultDocumentationCollector$3.compute(DocumentationManager.java:1108)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:888)
	at com.intellij.codeInsight.documentation.DocumentationManager$DefaultDocumentationCollector.getDocumentation(DocumentationManager.java:1107)
	at com.intellij.codeInsight.documentation.DocumentationManager$9.run(DocumentationManager.java:700)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:351)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:439)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at com.intellij.util.concurrency.QueueProcessor$RunnableConsumer.consume(QueueProcessor.java:298)
	at com.intellij.util.concurrency.QueueProcessor$RunnableConsumer.consume(QueueProcessor.java:295)
	at com.intellij.util.concurrency.QueueProcessor$2$1.run(QueueProcessor.java:110)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.concurrency.QueueProcessor$2.consume(QueueProcessor.java:107)
	at com.intellij.util.concurrency.QueueProcessor$2.consume(QueueProcessor.java:104)
	at com.intellij.util.concurrency.QueueProcessor$3$1.run(QueueProcessor.java:215)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.concurrency.QueueProcessor$3.run(QueueProcessor.java:212)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:400)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:439)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918)
	at java.lang.Thread.run(Thread.java:695)
	at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:56)
Caused by: java.util.InputMismatchException
	at java.util.Scanner.throwFor(Scanner.java:840)
	at java.util.Scanner.next(Scanner.java:1461)
	at java.util.Scanner.nextInt(Scanner.java:2091)
	at java.util.Scanner.nextInt(Scanner.java:2050)
	at com.haskforce.highlighting.annotation.external.GhcUtil.handleTypeInfo(GhcUtil.java:39)
	at com.haskforce.highlighting.annotation.external.GhcModi$2.call(GhcModi.java:103)
	at com.haskforce.highlighting.annotation.external.GhcModi$2.call(GhcModi.java:99)
	at com.haskforce.highlighting.annotation.external.GhcModi$4.call(GhcModi.java:318)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918)
	at java.lang.Thread.run(Thread.java:695)
```

**Comment by carymrobbins** [comment]
That's odd that you are seeing an `InputMismatchException` since that should be resolved as of version `0.3-beta.25`.  Are you running that version or an earlier one?  I'll be making a release here shortly, so you might as well update to the latest version anyway.

Also, check your build settings in **Project Preferences > Build, Execution, Deployment > Compiler > Haskell Compiler**.  Are you using stack or cabal for building?  Currently HaskForce is able to do some PATH hacking to show ghc-mod where your ghc is, but that hack may not work with stack currently (although, we could certainly get it to work by checking `stack path`).

**Comment by carymrobbins** [comment]
@Daenyth - I believe this issue should be resolved (as per my previous comment).  Please reopen if the bug persists.
